e2e-test/src/test/features/fileplugin/source/FileToBigQuery.feature

```diff
@@ -0,0 +1,411 @@
+    Then Save the pipeline
+    Then Preview and run the pipeline
+    Then Wait till pipeline preview is in running state
+    Then Open and capture pipeline preview logs
+    Then Verify the preview run status of pipeline in the logs is "succeeded"
+    Then Close the pipeline logs
+    Then Click on the Preview Data link on the Sink plugin node: "BigQuery"
+    Then Verify sink plugin's Preview Data for Input Records table and the Input Schema matches the Output Schema of Source plugin
+    Then Close the preview data
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Close the pipeline logs
+    Then Validate OUT record count is equal to records transferred to target BigQuery table
+
+  @BQ_SINK_TEST
+  Scenario: To verify data is getting transferred from File source to BigQuery with file properties as macro arguments
+    Given Open Datafusion Project to configure pipeline
+    When Select plugin: "File" from the plugins list as: "Source"
+    When Expand Plugin group in the LHS plugins list: "Sink"
+    When Select plugin: "BigQuery" from the plugins list as: "Sink"
+    Then Connect plugins: "File" and "BigQuery" to establish connection
+    Then Navigate to the properties page of plugin: "File"
+    Then Enter input plugin property: "referenceName" with value: "FileReferenceName"
+    Then Click on the Macro button of Property: "path" and set the value to: "fileSourcePath"
+    Then Click on the Macro button of Property: "formatMacroInput" and set the value to: "fileSourceFormat"
+    Then Click on the Macro button of Property: "skipHeaderMacroInput" and set the value to: "fileSourceSkipHeader"
+    Then Select Macro action of output schema property: "outputSchemaMacroInput" and set the value to "fileSourceOutputSchema"
+    Then Validate "File" plugin properties
+    Then Close the Plugin Properties page
+    Then Navigate to the properties page of plugin: "BigQuery"
+    Then Replace input plugin property: "projectId" with value: "projectId"
+    Then Enter input plugin property: "datasetProjectId" with value: "projectId"
+    Then Enter input plugin property: "referenceName" with value: "BQReferenceName"
+    Then Enter input plugin property: "dataset" with value: "dataset"
+    Then Enter input plugin property: "table" with value: "bqTargetTable"
+    Then Click plugin property: "truncateTable"
+    Then Click plugin property: "updateTableSchema"
+    Then Validate "BigQuery" plugin properties
+    Then Close the Plugin Properties page
+    Then Save the pipeline
+    Then Preview and run the pipeline
+    Then Enter runtime argument value "csvFile" for key "fileSourcePath"
+    Then Enter runtime argument value "csvFormat" for key "fileSourceFormat"
+    Then Enter runtime argument value "skipHeaderTrue" for key "fileSourceSkipHeader"
```

Comment:
can we have a test case where skipHeader is set to false?

Reply:
Added scenario with skipHeader set to false